src/html5_parser/parser/adoption_agency.rs

```diff
@@ -340,43 +340,43 @@ mod test {
 
         parser
             .active_formatting_elements
-            .push(ActiveElement::Node(NodeId(3)));
+            .push(ActiveElement::Node(3.into()));
```

Comment:
You can also use `NodeId::from(3)`:
```rust
.push(ActiveElement::Node(NodeId::from(3)));
```

Reply:
With these changes, it is possible to remove `pub(crate)` from the inner field of `NodeId`:
```rust
pub struct NodeId(usize);
```

But I didn't do this because it's not urgent, and because I didn't want to undo https://github.com/gosub-browser/gosub-engine/pull/100.

Reply:
What are the differences between `NodeId(X)`, `X.into()` and `NodeId::from(X)`? Are these mainly readability or is there overhead with the current `NodeId(X)`?

Reply:
`NodeId(X)` can only be done if you're in the module, or the inner field is `pub` or `pub(crate)`.  `X.into()` and `NodeId::from(X)` are equivalent, and both become available if you implement [From<usize>](https://github.com/gosub-browser/gosub-engine/blob/6fb63ba7083b0280224e84b33b7560ca7af260a4/src/html5_parser/node.rs#L49-L53) in this case.

Reply:
oh ok, got it

Reply:
I believe the Rust compiler is smart enough to know that you're using `NodeId` as a `usize` and will optimize it away.  The [newtype pattern](https://rust-unofficial.github.io/patterns/patterns/behavioural/newtype.html) is pretty common in Rust and is used to take ownership of external types so that you can use Rust's type system to ensure that you're not passing something that shouldn't be a node id into a function argument.